File: old_versions/ProgAndPlay/Client_Interfaces/c-cpp/exemple/projet/mission_6.c
#include "PP_Client.h"
#include "constantList_KP4.1.h"

/* Move all units to a specific position */
int main (){
	int i;
	/* define target position */
	PP_Pos p;
	p.x = 256.0;
	p.y = 811.0;
	PP_Open(); /* open the Prog&Play API */
	/* iterate all units */
	for (i = 0 ; i < PP_GetNumUnits (MY_COALITION) ; i++){
		if (PP_Unit_GetType(PP_GetUnitAt(MY_COALITION,i)) == ASSEMBLER)
			PP_Unit_ActionOnPosition(PP_GetUnitAt(MY_COALITION, i), MOVE, p);
	}
	/* close the Prog&Play API */
	PP_Close();
	return 0;
}

File: old_versions/ProgAndPlay/Client_Interfaces/python/mission6.py
from constantList_KP4_1 import *
from pp import *

# Définition de la cible
pos = PP_Pos(256.0, 811.0)
# Ouverture de la connexion avec le jeu
PP_Open()
# Parcourir toutes les unités
i = 0
while i < PP_GetNumUnits(MY_COALITION):
    # Récupérer l'unité courante
    u = PP_GetUnitAt(MY_COALITION, i)
    # Donner l'ordre de déplacement
    if PP_Unit_GetType(u) == ASSEMBLER:
        PP_Unit_ActionOnPosition(u, MOVE, pos)
    i += 1
# Fermer la connexion avec le jeu
PP_Close()
